### PR TITLE
Google map refactor

### PR DESCRIPTION
**Pull request by Siegrift**
Rewrite google map component to react + redux. 
Table has limited functionality, but is much more performant.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,26 @@
...
+
+#### prepojenia
+Kód [https://verejne.digital/prepojenia/index.html](https://verejne.digital/prepojenia/index.html).
+V priečinku `client` sa nachádza kód frontendu. Zvyšok priečinka tvorí backend.
+
+#### profil
+Kód [https://verejne.digital/profil/](https://verejne.digital/profil/).
+V priečinku `client` sa nachádza kód frontendu.
+
+Na developovanie react-u treba mať nainštalovaný `Node.js` a `npm`. Pred spustením, treba nainštalovať potrebné balíčky: `npm i`. Frontend spustíte pomocou `npm run start`.
```
Comment (tomaskulich) [inline README.md]: preco nie yarn? Zatial to vyzera byt npm vs yarn agnostic, preco nezobrat lepsiu technologiu?

File: README.md [hunk 2]
```diff
@@ -1,4 +1,26 @@
...
+
+## Backend
```
Comment (tomaskulich) [inline README.md]: Don't create TODOs for other than your internal need.

File: static/client/src/utils/helpers.js [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+  }
+)
+
+/*
+ * Lookups value on a specific path in a given state. If the lookup fails, the error is thrown
+ * unless default value is specified. Last argument is a map of default values:
+ * - last is used when access fails at last step
+ * - any is used when access fails at any step
+ */
+export const getIn = (state, path, {last, any} = {}) => {
```
Comment (tomaskulich) [inline static/client/src/utils/helpers.js]: Use lodash.get - don't use custom functions unless necessary

File: static/client/src/utils/helpers.js [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+      }
+    }
+  }
+  return value
+}
+
+/*
+ * Checks whether path exists in a given state.
+ */
+export const hasIn = (state, path) => {
```
Comment (tomaskulich) [inline static/client/src/utils/helpers.js]: detto

File: static/client/webpack/config.base.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
+import path from 'path'
```
Comment (tomaskulich) [inline static/client/webpack/config.base.js]: Why custom webpack config? What's wrong about webpack-config-vacuumlabs?

File: static/client/src/App/actions.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import { PATH_APP } from './state'
+
+export const toggleButtonText = () => ({
+  type: 'Toggle button text',
+  path: [...PATH_APP, 'buttonText'],
```
Comment (tomaskulich) [inline static/client/src/App/actions.js]: Drop the 'path' concept. With immutability-helper, it buys us nothing.

File: static/client/src/utils/helpers.js [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+/*
+ * Sets specific `path` of connected component state using uiDispatch.
+ *
+ * component: component decorated by connect
+ * path: array of string keys
+ * val: new value for the `path`
+ * msg: dispatch message
+ * args: dispatch arguments
+ */
+export function setInLocal(component, msg, path, val) {
```
Comment (tomaskulich) [inline static/client/src/utils/helpers.js]: remove

File: static/client/.eslintrc [hunk 8]
```diff
@@ -0,0 +1,196 @@
+{
```
Comment (tomaskulich) [inline static/client/.eslintrc]: this shouldn't be needed at all.

File: static/client/src/api/index.js [hunk 9]
```diff
@@ -0,0 +1,8 @@
+class Api {
```
[the remark this answers is not in the document]
Comment (Siegrift) [reply static/client/src/api/index.js]: I would like the to add more complexity to Api class (e.g. cache some requests), or maybe even use it like https://github.com/vacuumlabs/MyPictures/blob/master/src/client/api/index.js
The point now is to seprate the api, and use it in thunk middle as extra argument.

File: static/client/webpack/config.base.js [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+  },
+  plugins: [
+    new webpack.DefinePlugin({
+      'process.env': {
+        NODE_ENV: `'${process.env.NODE_ENV}'`,
+      },
+    }),
+    new webpack.ProvidePlugin({
+      lodash: 'lodash',
+      Promise: 'bluebird',
```
[the remark this answers is not in the document]
Comment (Siegrift) [reply static/client/webpack/config.base.js]: Hej, ale itak Bluebird harcodujem v index.js a lodash vzdy importujem...

File: static/client/src/utils/helpers.js (hunk 3 above)
Comment (Siegrift) [reply static/client/src/utils/helpers.js]: O tomto som sa bavil raz so samom sladkom, presne som sa ho nato pytal, ze preco nepouzivat tie lodashovske. Reason je ze tento nas (pershingov) ti da lepsi error message (+tusim sa este inak chova, ale nepamatam kde).

**Pull request closed without merging**